src/api/cpp/array.cpp

```diff
@@ -949,6 +949,8 @@ INSTANTIATE(long long)
 INSTANTIATE(unsigned long long)
 INSTANTIATE(short)
 INSTANTIATE(unsigned short)
+INSTANTIATE(af_half)
+INSTANTIATE(half_float::half)
```

Comment:
Can we create arrays of type `half_float::half` type as well ? I was under the impression that `half_float::float` is used only in our unit tests.